docs/Easystack-files.rst

```diff
@@ -32,68 +39,80 @@ To build software with *Easystack*, type:
 
 where ``example.yaml`` is the file with specifications that you just created (more about this in the next section).
 
+
 .. _easystack_structure:
 
 Structure of an easystack file
 ------------------------------
 
 Easystack files are written in `YAML syntax <https://learnxinyminutes.com/docs/yaml>`_.
 
-Essentially, easystack files list the EasyConfigs you want to install. These are listed under the 'easyconfigs:' key:
+Essentially, an easystack file lists the easyconfig files you want to install, which are specified under the
+`easyconfigs` key.
+
+For example:
 
 .. code:: yaml
 
   easyconfigs:
     - PyTorch-1.12.0-foss-2022a-CUDA-11.7.0.eb
     - OpenFOAM-v2206-foss-2022a.eb
 
-(note the use of the ``-`` to list the items).
+.. note::
+   You must use '``-``' to list the easyconfigs in an easystack file.
+
+In addition, you can specify additional configuration options via the ``options`` subkey
+which will only apply to the installation of a particular easyconfig file.
 
-In addition, you can specify additional options that will only apply to the installation of that EasyConfig:
+For example:
 
 .. code:: yaml
 
   easyconfigs:
     - PyTorch-1.12.0-foss-2022a-CUDA-11.7.0.eb:
-      options: {
-        'from-pr': 15924,
-        'debug': True
-      }
+      options:
+        from-pr: 15924
+        debug: True
     - Hypre-2.25.0-foss-2022a.eb:
     - OpenFOAM-v2206-foss-2022a.eb:
-      options: {
-        'installpath': '/my/custom/installpath'
-      }
+      options:
+        installpath: /my/custom/installpath
+
+.. note::
+   You need to take care with some values in YAML, especially integers, booleans, etc.
+   To avoid that the YAML parser assumes that a value is of a particular type,
+   use quotes (``'...'``) to force a string value.
```

Comment:
This comment confuses me: after reading it, I don't know when to use quotes, and when not to. Our examples don't use quotes for the PR number. Should we have? I think if we put this note, it needs more explaination of when to use quotes, and when not too.

Reply:
OK, I'll try to clarify (you should use quotes if the value definitely needs to be a string value, for `--from-pr` it's OK if it's an integer)